packages/app/background-jobs-common/src/background-job-processor/processors/AbstractBackgroundJobProcessor.ts

```diff
@@ -455,13 +482,21 @@ export abstract class AbstractBackgroundJobProcessor<
    * @param job
    * @protected
    */
-  protected async purgeJobData(job: JobType): Promise<void> {
+  async purgeJobData(job: JobType): Promise<void> {
     const jobOptsRemoveOnComplete = job.opts.removeOnComplete
     if (jobOptsRemoveOnComplete === true || jobOptsRemoveOnComplete === 1) return
 
     // @ts-ignore
-    await job.updateData({ metadata: job.data.metadata })
-    await job.clearLogs()
+    const updateDataPromise = job.updateData({ metadata: job.data.metadata })
+    const clearLogsPromise = job.clearLogs()
+
+    const updateDataPromiseIndex = this.runningPromises.push(updateDataPromise) - 1
+    const clearLogsPromiseIndex = this.runningPromises.push(clearLogsPromise) - 1
+
+    await Promise.all([updateDataPromise, clearLogsPromise]).then(() => {
+      this.runningPromises.splice(updateDataPromiseIndex, 1)
+      this.runningPromises.splice(clearLogsPromiseIndex, 1)
+    })
```

Comment:
Good idea! 🚀

Reply:
Copyright @kibertoad 😉

Reply:
it doesn't catch all cases, though, I still want to investigate if it's possible to somehow force Redis connection to drain all ongoing operations, then we wouldn't need to manage it ourselves. but that'd be outside of this PR